File: Rule Engine/Services/Axis.Service.Admin/Admin/AdminService.cs
using System.Diagnostics;
using Axis.Configuration;
using Axis.Model.Account;
using Axis.Model.Common;
using Axis.Model.Security;
using Axis.Security;
using Axis.Service;
using System;
using System.Collections.Generic;
using System.Collections.Specialized;
using System.Globalization;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Web;

namespace Axis.Service.Admin
{
    public class AdminService : IAdminService
    {
        #region Class Variables

        private CommunicationManager communicationManager;
        private const string baseRoute = "admin/";

        #endregion

        #region Constructors

        public AdminService()
        {
            communicationManager = new CommunicationManager(ApplicationSettings.Token, AuthContext.Auth.Token.Token);
        }

        #endregion

        #region Methods

        public Response<UserModel> GetUsers(string userSch)
        {
            var apiUrl = baseRoute + "GetUsers";
            var param = new NameValueCollection();
            param.Add("userSch", string.IsNullOrEmpty(userSch) ? string.Empty : userSch);

            return communicationManager.Get<Response<UserModel>>(param, apiUrl);
        }

        public Response<UserModel> AddUser(UserModel user)
        {
            var apiUrl = baseRoute + "AddUser";
            return communicationManager.Post<UserModel, Response<UserModel>>(user, apiUrl);
        }

        public Response<UserModel> UpdateUser(UserModel user)
        {
            var apiUrl = baseRoute + "UpdateUser";
            return communicationManager.Post<UserModel, Response<UserModel>>(user, apiUrl);
        }

        public Response<UserModel> RemoveUser(int userID, DateTime modifiedOn)
        {
            var apiUrl = baseRoute + "RemoveUser";
            var param = new NameValueCollection
            {
                {"userID", userID.ToString()},
                {"modifiedOn", modifiedOn.ToString(CultureInfo.InvariantCulture)}
            };
            return communicationManager.Delete<Response<UserModel>>(param, apiUrl);
        }

        public Response<UserModel> ActivateUser(UserModel user)
        {
            var apiUrl = baseRoute + "ActivateUser";
            return communicationManager.Post<UserModel, Response<UserModel>>(user, apiUrl);
        }

        public Response<UserModel> UnlockUser(UserModel user)
        {
            var apiUrl = baseRoute + "UnlockUser";
            return communicationManager.Post<UserModel, Response<UserModel>>(user, apiUrl);
        }

        public Response<UserModel> SendNewUserEmail(UserModel user)
        {
            var apiUrl = baseRoute + "SendNewUserEmail";
            return communicationManager.Post<UserModel, Response<UserModel>>(user, apiUrl);
        }

        public Response<UserRoleModel> GetUserRoles(int userID)
        {
            var apiUrl = baseRoute + "GetUserRoles";
            var param = new NameValueCollection();
            param.Add("userID", userID.ToString());
            return communicationManager.Get<Response<UserRoleModel>>(param, apiUrl);
        }

        public Response<UserCredentialModel> GetUserCredentials(int userID)
        {
            var apiUrl = baseRoute + "GetUserCredentials";
            var param = new NameValueCollection();
            param.Add("userID", userID.ToString());
            return communicationManager.Get<Response<UserCredentialModel>>(param, apiUrl);
        }

        public Response<UserCredentialModel> AddUserCredential(UserCredentialModel userCredential)
        {
            var apiUrl = baseRoute + "AddUserCredential";
            return communicationManager.Post<UserCredentialModel, Response<UserCredentialModel>>(userCredential, apiUrl);
        }

        public Response<UserCredentialModel> UpdateUserCredential(UserCredentialModel userCredential)
        {
            var apiUrl = baseRoute + "UpdateUserCredential";
            var param = new NameValueCollection();
            return communicationManager.Post<UserCredentialModel, Response<UserCredentialModel>>(userCredential, apiUrl);
        }

        public Response<UserCredentialModel> DeleteUserCredential(int userCredentialID, DateTime modifiedOn)
        {
            var apiUrl = baseRoute + "DeleteUserCredential";
            var param = new NameValueCollection
            {
                {"userCredentialID", userCredentialID.ToString()},
                {"modifiedOn", modifiedOn.ToString(CultureInfo.InvariantCulture)}
            };
            return communicationManager.Delete<Response<UserCredentialModel>>(param, apiUrl);
        }

        public Response<ResetPasswordModel> VerifySecurityDetails(ResetPasswordModel resetPassword)
        {
            var apiUrl = baseRoute + "verifySecurityDetails";

            return communicationManager.Post<ResetPasswordModel, Response<ResetPasswordModel>>(resetPassword, apiUrl);
        }

        #endregion
    }
}

File: Presentation Engine/Axis.PresentationEngine/Areas/Admin/Scripts/app/services/adminService.js
angular.module('xenatixApp')
    .factory('adminService', ["$http", "$q", 'settings', function ($http, $q, settings) {
        var CONFIG = {
            controllerAction: "/admin/admin/"
        };
        function get(userSch) {
            var dfd = $q.defer();
            $http.get(settings.webApiBaseUrl + CONFIG.controllerAction + 'GetUsers', { params: { userSch: userSch } })
            .success(function (data, status, header, config) {
                dfd.resolve(data);
            })
            .error(function (data, status, header, config) {
                dfd.reject(status);
            });

            return dfd.promise;
        };

        function add(user) {
            var dfd = $q.defer();
            $http.post(settings.webApiBaseUrl + CONFIG.controllerAction + 'AddUser', user)
            .success(function (data, status, header, config) {
                dfd.resolve(data);
            })
            .error(function (data, status, header, config) {
                dfd.reject(status);
            });

            return dfd.promise;
        }

        function update(user) {
            var dfd = $q.defer();
            
            $http.post(settings.webApiBaseUrl + CONFIG.controllerAction + 'UpdateUser', user)
            .success(function (data, status, header, config) {
                dfd.resolve(data);
            })
            .error(function (data, status, header, config) {
                dfd.reject(status);
            });

            return dfd.promise;
        }

        function remove(user) {
            var dfd = $q.defer();
            user.ModifiedOn = moment.utc();
            $http.post(settings.webApiBaseUrl + CONFIG.controllerAction + 'RemoveUser', user)
            .success(function (data, status, header, config) {
                dfd.resolve(data);
            })
            .error(function (data, status, header, config) {
                dfd.reject(status);
            });
            
            return dfd.promise;
        }

        function activate(user) {
            var dfd = $q.defer();

            $http.post(settings.webApiBaseUrl + CONFIG.controllerAction + 'ActivateUser', user)
            .success(function (data, status, header, config) {
                dfd.resolve(data);
            })
            .error(function (data, status, header, config) {
                dfd.reject(status);
            });

            return dfd.promise;
        }

        function reset(primaryEmail) {
            var dfd = $q.defer();
            var resetPasswordControllerAction = '/account/forgotPassword/';

            $http.post(settings.webApiBaseUrl + resetPasswordControllerAction + 'sendResetLink', { email: primaryEmail })
            .success(function (data, status, header, config) {
                dfd.resolve(data);
            })
            .error(function (data, status, header, config) {
                dfd.reject(status);
            });

            return dfd.promise;
        }

        function unlock(user) {
            var dfd = $q.defer();

            $http.post(settings.webApiBaseUrl + CONFIG.controllerAction + 'UnlockUser', user)
            .success(function (data, status, header, config) {
                dfd.resolve(data);
            })
            .error(function (data, status, header, config) {
                dfd.reject(status);
            });

            return dfd.promise;
        }

        function getUserRoles(user) {
            var dfd = $q.defer();

            $http.get(settings.webApiBaseUrl + CONFIG.controllerAction + 'GetUserRoles', { params: { userID: user.UserID } })
            .success(function (data, status, header, config) {
                dfd.resolve(data);
            })
            .error(function (data, status, header, config) {
                dfd.reject(status);
            });

            return dfd.promise;
        }

        function getUserCredentials(user) {
            var dfd = $q.defer();

            $http.post(settings.webApiBaseUrl + CONFIG.controllerAction + 'GetUserCredentials', user)
            .success(function (data, status, header, config) {
                dfd.resolve(data);
            })
            .error(function (data, status, header, config) {
                dfd.reject(status);
            });

            return dfd.promise;
        }

        function addUserCredential(userCredential) {
            var dfd = $q.defer();

            $http.post(settings.webApiBaseUrl + CONFIG.controllerAction + 'AddUserCredential', userCredential)
            .success(function (data, status, header, config) {
                dfd.resolve(data);
            })
            .error(function (data, status, header, config) {
                dfd.reject(status);
            });

            return dfd.promise;
        }

        function updateUserCredential(userCredential) {
            var dfd = $q.defer();

            $http.post(settings.webApiBaseUrl + CONFIG.controllerAction + 'UpdateUserCredential', userCredential)
            .success(function (data, status, header, config) {
                dfd.resolve(data);
            })
            .error(function (data, status, header, config) {
                dfd.reject(status);
            });

            return dfd.promise;
        }

        function removeUserCredential(userCredentialID) {
            var dfd = $q.defer();

            $http.delete(settings.webApiBaseUrl + CONFIG.controllerAction + 'DeleteUserCredential',
                { params: { userCredentialID: userCredentialID, modifiedOn: moment.utc().format() } })
            .success(function (data, status, header, config) {
                dfd.resolve(data);
            })
            .error(function (data, status, header, config) {
                dfd.reject(status);
            });

            return dfd.promise;
        }

        return {
            get: get,
            add: add,
            update: update,
            remove: remove,
            reset: reset,
            unlock: unlock,
            getUserRoles: getUserRoles,
            getUserCredentials: getUserCredentials,
            addUserCredential: addUserCredential,
            updateUserCredential: updateUserCredential,
            removeUserCredential: removeUserCredential,
            activate: activate
        };
    }]);
